+++ a/lib/python2.7/dist-packages/xdot/xdot.py	2015-04-21 08:10:36.903315281 +0000
+++ b/lib/python2.7/dist-packages/xdot/xdot.py	2015-04-21 08:10:36.903315281 +0000
@@ -477,7 +477,7 @@
         return res
 
     def read_number(self):
-        return int(self.read_code())
+        return int(self.read_float())
 
     def read_float(self):
         return float(self.read_code())
